src/generator.ts

```diff
@@ -0,0 +1,183 @@
+    this.logger.debug(`Templates directory: ${this.templatesDir}`);
+  }
+
+  public get cwd(): string {
+    return this.workingDir;
+  }
+
+  public set cwd(value: string) {
+    this.workingDir = value;
+  }
+
+  public async render(source: string, destination: string, data?: Record<string, unknown>): Promise<void> {
+    const fullSource = join(this.templatesDir, source);
+    const dryRunMsg = this.dryRun ? '[DRY RUN] ' : '';
+    this.logger.debug(`${dryRunMsg}Rendering template ${fullSource} to ${destination}`);
+
+    const rendered = await new Promise<string>((resolve, reject) => {
+      renderFile(fullSource, data ?? {}, (err, str) => {
+        if (err) reject(err);
+        return resolve(str);
+      });
+    });
+
+    let verb = 'Creating';
+    if (rendered) {
+      const relativePath = relative(process.cwd(), destination);
+      if (await fileExists(destination)) {
+        const confirmation =
+          this.force ??
+          (await (
+            await import('@inquirer/confirm')
+          ).default({
+            message: `Overwrite ${relativePath}?`,
+          }));
+        if (confirmation) {
+          verb = 'Overwriting';
+        } else {
+          this.ux.log(`${dryRunMsg}${colorize('yellow', 'Skipping')} ${relativePath}`);
+          return;
+        }
+      }
+
+      this.ux.log(`${dryRunMsg}${colorize('yellow', verb)} ${relativePath}`);
+
+      if (!this.dryRun) {
+        await mkdir(dirname(destination), { recursive: true });
+        await writeFile(destination, rendered);
+      }
+    }
+  }
+
+  public execute(cmd: string): void {
+    if (this.dryRun) {
+      this.ux.log(`[DRY RUN] ${cmd}`);
+      return;
+    }
+
+    this.logger.debug(`Executing command: ${cmd}`);
+    shelljs.exec(cmd, { cwd: this.cwd });
+  }
+
+  public async loadPjson(): Promise<PackageJson> {
+    try {
+      this.pjson = JSON.parse(await readFile(join(this.cwd, 'package.json'), 'utf-8')) as PackageJson;
+    } catch {
+      if (this.dryRun) {
+        this.pjson = {
+          name: '',
+          description: '',
+          dependencies: {},
+          devDependencies: {},
+          files: [],
+          bugs: '',
+          homepage: '',
+          repository: '',
+          oclif: {
+            bin: '',
+            topics: {},
+            dirname: '',
+          },
+          author: '',
+          // @ts-expect-error - not all properties are required
+          scripts: {},
+          // @ts-expect-error - not all properties are required
+          wireit: {},
+        };
+      }
+    }
+
+    return this.pjson;
+  }
+
+  public async writePjson(): Promise<void> {
+    const updating = colorize('yellow', 'Updating');
+    if (this.dryRun) {
+      this.ux.log(`[DRY RUN] ${updating} package.json`);
+      return;
+    }
+
+    this.ux.log(`${updating} package.json`);
+    await writeFile(join(this.cwd, 'package.json'), JSON.stringify(this.pjson, null, 2));
```

Comment:
we had to `normalize` this path to fix a bug [recently](https://github.com/salesforcecli/plugin-dev/pull/515/files) might need to be carried over here too